src/components/ProtectedPage.jsx

```diff
@@ -20,7 +20,7 @@ const ProtectedPage = ({ title, children, restrictions }) => {
       !session.user.role
     ) {
       console.log("Have not register");
-      router.push("/");
+      setUnauthorizedError("You need login to access this page.");
```

Comment:
keep track of the status code since that changes as well, u can store it as an object